src/main/java/com/auth0/jwk/Jwk.java

```diff
@@ -159,26 +169,65 @@ public String getCertificateThumbprint() {
-        } catch (NoSuchAlgorithmException e) {
-            throw new InvalidPublicKeyException("Invalid algorithm to generate key", e);
+        PublicKey publicKey = null;
+
+        switch (type) {
+            case ALGORITHM_RSA:
+                try {
+                    KeyFactory kf = KeyFactory.getInstance(ALGORITHM_RSA);
+                    BigInteger modulus = new BigInteger(1, Base64.decodeBase64(stringValue("n")));
+                    BigInteger exponent = new BigInteger(1, Base64.decodeBase64(stringValue("e")));
+                    publicKey = kf.generatePublic(new RSAPublicKeySpec(modulus, exponent));
+                } catch (InvalidKeySpecException e) {
+                    throw new InvalidPublicKeyException("Invalid public key", e);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new InvalidPublicKeyException("Invalid algorithm to generate key", e);
+                }
+                break;
+
+            case ALGORITHM_ELLIPTIC_CURVE:
+                try {
+                    KeyFactory keyFactory = KeyFactory.getInstance(ALGORITHM_ELLIPTIC_CURVE);
+                    ECPoint ecPoint = new ECPoint(new BigInteger(Base64.decodeBase64(stringValue("x"))),
+                            new BigInteger(Base64.decodeBase64(stringValue("y"))));
+                    AlgorithmParameters algorithmParameters = AlgorithmParameters.getInstance(ALGORITHM_ELLIPTIC_CURVE);
+                    switch (stringValue("crv")) {
+                        case ELLIPTIC_CURVE_TYPE_P256:
+                            algorithmParameters.init(new ECGenParameterSpec("secp256r1"));
+                            break;
+                        case ELLIPTIC_CURVE_TYPE_P384:
+                            algorithmParameters.init(new ECGenParameterSpec("secp384r1"));
+                            break;
+                        case ELLIPTIC_CURVE_TYPE_P521:
+                            algorithmParameters.init(new ECGenParameterSpec("secp521r1"));
+                            break;
+                        default:
+                            throw new InvalidPublicKeyException("Invalid or unsupported curve type " + stringValue("crv"));
+                    }
+                    ECParameterSpec ecParameterSpec = algorithmParameters.getParameterSpec(ECParameterSpec.class);
+                    ECPublicKeySpec ecPublicKeySpec = new ECPublicKeySpec(ecPoint, ecParameterSpec);
+                    publicKey = keyFactory.generatePublic(ecPublicKeySpec);
+                } catch (InvalidParameterSpecException e) {
+                    throw new InvalidPublicKeyException("Invalid public key", e);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new InvalidPublicKeyException("Invalid algorithm to generate key", e);
+                } catch (InvalidKeySpecException e) {
+                    throw new InvalidPublicKeyException("Invalid public key", e);
```

Comment:
@lbalmaceda I tried to write a test to trigger these checked exceptions, but unfortunately my runtime environment only throws `RuntimeException`s despite the API advertising checked exception types `InvalidKeySpecException` and `InvalidParameterSpecException`.  I'm not sure how to force these short of maybe trying to manipulate the JCE provider with some crazy object proxies at runtime -- probably not an approach worth taking.  

If you have any suggestions on how to force these exceptions, I'm all ears. :)  Otherwise I'm comfortable if you are  with missing coverage for these degenerate cases given that they're both unrecoverable and uncontrollable.

Reply:
I've raised this specific query for internal discussion and we agreed that if the exceptions are being re-thrown there's not much value in covering these.